packages/VendorLocker/ParentClassMethodTypeOverrideGuard.php

```diff
@@ -20,30 +23,60 @@ public function __construct(
         private readonly ReflectionResolver $reflectionResolver,
         private readonly TypeComparator $typeComparator,
         private readonly StaticTypeMapper $staticTypeMapper,
+        private readonly PrivatesAccessor $privatesAccessor,
     ) {
     }
 
-    public function hasParentClassMethod(ClassMethod $classMethod): bool
+    public function hasParentClassMethod(ClassMethod $classMethod): ?bool
     {
-        return $this->getParentClassMethod($classMethod) instanceof MethodReflection;
+        try {
+            $parentClassMethod = $this->resolveParentClassMethod($classMethod);
+
+            return $parentClassMethod instanceof MethodReflection;
+        } catch (UnresolvableClassException) {
+            // we don't know all involved parents.
+            return null;
```

Comment:
just return true here, so it "marked" as has parent so `bool` can be returned.

Reply:
I don't think its  a good idea to handle "we don't know" the same as "sure, the method exists".

for our callers, as they exist right now, this is acceptable, but it might suprisingly break in the future as soon as someone compares the result of this method against `true`

Reply:
I prefer to create separate service for param guard handling, eg: `ClassMethodParamTypeOverrideGuard`.

Something like `ClassMethodReturnTypeOverrideGuard::shouldSkipClassMethod()` do

https://github.com/rectorphp/rector-src/blob/73112a253e40199adda14259717e4d8c630c97d1/packages/VendorLocker/NodeVendorLocker/ClassMethodReturnTypeOverrideGuard.php#L51

Reply:
@staabm just checked the `getParentClassMethod()` usages in rules, there are various usage directly, which cause exception, I think try catch should be in consumer instead of in `hasParentClassMethod()` as that cause double call.

I will create new PR for that.

Reply:
the 3 consumer I could find, could all use `hasParentclassMethod()` instead of `getParentClassMethod()`, which would also make the code more readable

Reply:
hm.., the service is named `ParentClassMethodTypeOverrideGuard`, which `has` vs `get` should only validate instead of getting in real use case, so I prefer return bool true for `hasParentClassMethod()`, and return null for `getParentClassMethod()` when no parent at all.

Reply:
I create PR 

- https://github.com/rectorphp/rector-src/pull/4703 for it, 

`return bool` only should be the way to go as it only validate, not a "real" verification, we then return `?MethodReflection` for `getParentClassMethod()`